### DISCUSSION TITLE
Toggling Checkbox on a ModalScreen when key is pressed

### DISCUSSION THREAD

**Discussion by msempere** [Textualize/textual#2238] [Q&A] [answered]
Hi! I'm playing with the recently added `ModalScreen` and the  `Checkbox` widget and I have noticed I can toggle a checkbox in the `on_mount` with `self.query_one("#checkbox-0", Checkbox).toggle()` but I cannot do the same as part of the `on_key`. Is that a expected behaviour?

The following code doesn't toggle the widget:
```python
def on_key(self, event: Key):
    self.query_one("#checkbox-0", Checkbox).toggle()
```

While this does:

```python
def on_mount(self):
    self.query_one("#checkbox-0", Checkbox).toggle()
```

**Comment by davep** [comment, marked as answer]
Do you think you could post a minimal working example of code that shows the issue you're seeing? I just tested with this:

```python
from textual.app     import App, ComposeResult
from textual.widgets import Header, Footer, Checkbox

class CBKeyToggleApp( App[ None ] ):

    CSS = """
    Screen {
        align: center middle;
    }
    """

    def compose( self ) -> ComposeResult:
        yield Header()
        yield Checkbox("This is a test Checkbox", id="toggle-me")
        yield Footer()

    def on_key(self) -> None:
        self.query_one("#toggle-me", Checkbox).toggle()

if __name__ == "__main__":
    CBKeyToggleApp().run()
```

and it toggles the `Checkbox` just fine on any key.

Could you also post details of the exact steps you're taking too? For example, I *can* reproduce the issue you're seeing if I:

1. Run the above code.
2. Hit <kbd>Tab</kbd> to focus the `Checkbox`
3. Press either <kbd>Space</kbd> or <kbd>Enter</kbd>

In this case the above would be the expected behaviour [because those are the key bindings for toggling a `Checkbox`](https://github.com/Textualize/textual/blob/79ebbff9338028b91128ecda9bf7e87e63a22012/src/textual/widgets/_toggle_button.py#L34) so having an `on_key` also toggle it and then having the user press either of those keys would result in a double toggle.

**Reply by msempere** [reply]
Mmm interesting. I don't know what I was doing wrong before but I can verify it's working now.
Out of curiosity now that you've mentioned the double toggle, is the a way to replace a parent binding? in this case, make use of `space` or `enter` without the toggle? or just by setting the `on_key` to use `space`, per example, overwrites the parent binding?

**Reply by davep** [reply]
What do you mean exactly when you say "replace a parent binding"? Are you creating a custom `Checkbox` by inheriting from it? Perhaps it would be helpful if you explained what you're trying to do rather then how you're trying to do it so I can better understand?

I think the thing that has me a wee be confused is, if `Checkbox` already toggles using the keys you want to be used to toggle a `Checkbox`, why would you implement capturing those keys to toggle a `Checkbox`?
